src/State/User/UserSlice.ts

```diff
@@ -0,0 +1,26 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { UserData } from "../../DataTypes/UserType";
+
+const initialState: UserData = {
+  username: "",
+  name: "",
+  email: "",
+  id: 0,
+  profile_image: "",
+  comments_count: 0,
+  posts_count: 0
+};
+
+const userSlice = createSlice({
+  name: "user",
+  initialState,
+  reducers: {
+    saveUserData: (state, action: PayloadAction<UserData>) => {
+      return { ...state, ...action.payload };
```

Comment:
its better to have this data inside an object userData 
const initialState= {
userData: {
    username: "",
    name: "",
    email: "",
    id: 0,
    profile_image: "",
    comments_count: 0,
    posts_count: 0
  }
}